aries-site/src/examples/templates/PageHeader/PageHeader.js

```diff
@@ -1,24 +1,67 @@
+import { useContext } from 'react';
 import PropTypes from 'prop-types';
-import { Box, Header, Heading, Paragraph } from 'grommet';
+import {
+  Box,
+  Button,
+  Header,
+  Heading,
+  Menu,
+  Paragraph,
+  ResponsiveContext,
+} from 'grommet';
+import { More } from 'grommet-icons';
 
-export const PageHeader = ({ title, subtitle, actions, ...rest }) => (
-  <Header {...rest}>
-    <Box>
-      <Heading level={1} margin="none" size="small">
-        {title}
-      </Heading>
-      {typeof subtitle === 'string' ? (
-        <Paragraph margin="none">{subtitle}</Paragraph>
-      ) : (
-        <>{subtitle}</>
-      )}
-    </Box>
-    {actions}
-  </Header>
-);
+export const PageHeader = ({
+  title,
+  subtitle,
+  actions,
+  contextualNav,
+  ...rest
+}) => {
+  const size = useContext(ResponsiveContext);
+
+  return (
+    <Header direction="column" align="stretch" gap="small" {...rest}>
+      <Box align="start">{contextualNav}</Box>
+      <Box align="start" direction="row" gap="small" justify="between">
+        <Box gap="xsmall">
+          <Heading level={1} margin="none" size="small">
+            {title}
+          </Heading>
+          {typeof subtitle === 'string' ? (
+            <Paragraph margin="none" size="large">
+              {subtitle}
+            </Paragraph>
+          ) : (
+            <>{subtitle}</>
+          )}
+        </Box>
+        <Box direction="row" gap="small">
+          {!['xsmall', 'small'].includes(size) ? (
+            actions.map(action => (
```

Comment:
What happens if I pass in an array of 5+ action items? Is there a limit? Should those in excess of the limit be put under an "overflow menu"?

Reply:
Another great design question which I posed to Vicky yesterday for her to consider from the design side. Yes, there should be a limit of how many are showed flat on the screen.

From the dev side, I was exploring an approach locally (but didn't push up) where the PageHeader is implemented as a Grid. The page-level actions are "smart" about their gridArea dimensions and when they are wider than their allowed space, they collapse into a menu.

Reply:
Yes, indeed great question @halocline. I haven't gotten to this section of the PageHeader. However, I imagine we'd give guidance that IF there are more than 4 action items or when on a smaller resolution such as mobile, the action items will then live in a menu item or and "overflow menu" (this is still tbd). But again, have not gotten to this.

Reply:
Just planning some seeds for when you get there, doesn't have to be answered now. If there are (for example) 4 action items, do:
1. All the action items collapse into a menu?
2. 3 of the action items collapse into but the primary action stays visible next to the Menu.
3. Some other thing?